src/main.rs

```diff
@@ -31,6 +42,27 @@ struct UsageError {
     message: String,
 }
 
+/// Parses the value of a flag that takes a duration.
+///
+/// At the moment, the only supported unit for the duration are seconds and the textual value
+/// must include the "ms" suffix to support adding extra units in the future if desired.
+fn parse_duration(value: &str) -> Result<Timespec, UsageError> {
+    if !value.ends_with(SECONDS_SUFFIX) {
+        let message = format!("invalid time specification {}: must end in 's'", value);
+        return Err(UsageError { message });
+    }
+    let value = &value[0..value.len() - SECONDS_SUFFIX.len()];
+    match value.parse::<u32>() {
+        Ok(sec) => {
+            Ok(Timespec { sec: i64::from(sec), nsec: 0 })
+        },
+        Err(e) => {
+            let message = format!("invalid time specification {}: {}", value, e);
+            Err(UsageError { message })
+        }
+    }
```

Comment:
Personally I have a slight preference for the following. However what you have is also readable.

```rust
value.parse::<u32>()
  .map(|sec| Timespec { sec: i64::from(sec), nsec: 0 })
  .map_err(|e| UsageError { message: format!("invalid time specification {}: {}", value, e) })
```

Reply:
Nice. Wasn't familiar with this idiom and I like that it's more concise.